### DISCUSSION TITLE
Issues with low API level and libc++

### DISCUSSION THREAD

**Discussion by espenkn** [android/ndk#2004] [General]
When experimenting with upgrading the NDK I ran into some issues like (for brevity):

```
error: no member named 'timespec_get' in the global namespace; did you mean 'timespec'?
error: no member named 'aligned_alloc' in the global namespace
```

In the headers it says they are not available until API level 29 but the NDK README says it should support API 21. It looks like this comes from some internal usage in the chrono library.

Is there anyway to use the newer NDK with API level 21 to work around this?

**Comment by enh-google** [comment]
@rprichard

**Comment by espenkn** [comment]
Can anyone shed some light on this?

**Comment by enh-google** [comment]
@rprichard --- in case it wasn't obvious (i had to remind myself just now), this seems to be a <chrono> question: "why can't i use <chrono> when targeting api level 21?".

**Comment by DanAlbert** [comment]
Can you elaborate? Without an example of the failing code I don't know where to begin.

**Comment by espenkn** [comment]
I finally had time to do some more research and the compilation issues was actually coming from another custom compiler we use in the pipeline. It was hard to see and I'm sorry for wasting your time.
